src/lib/storage/index/abstract_table_index.hpp

```diff
@@ -79,68 +76,82 @@ class AbstractTableIndex : private Noncopyable {
 
   AbstractTableIndex() = delete;
   explicit AbstractTableIndex(const TableIndexType type);
-  AbstractTableIndex(AbstractTableIndex&&) = default;
+  AbstractTableIndex(AbstractTableIndex&&) = delete;
   virtual ~AbstractTableIndex() = default;
 
   /**
-   * Searches for all positions of the entry within the table index.
-   *
-   * Calls _equals() of the derived class.
-   * @param value used to query the index.
-   * @return A pair of Iterators containing the start and end iterator for the stored RowIDs of the element inside the
-   * table index.
+   * The following four methods are used to access any table index. Each of them accepts a generic function object (or
+   * similar) that expects a begin and end iterator to the underlying data structure as parameter. When accessing the
+   * index with one of these methods, one or two pairs of iterators are passed to the function object. For more
+   * information, please refer to the method description.
+   * 
+   * Unfortunately, it is not possible to locate the implementations of these methods inside the source file because of
+   * the separate compilation model. In this case, it is not possible to use explicit template instantiation since
+   * otherwise all used function objects and functors would have to be instantiated in abstract_table_index.cpp.
```

Comment:
Do you see any difference between function objects and functors? I assumed they were synonyms.

A quick google search query showed this:
>Functors (Function Objects or Functionals) are simply put object + ().

>In other words, a functor is any object that can be used with () in the manner of a function.

https://www.bogotobogo.com/cplusplus/functors.php